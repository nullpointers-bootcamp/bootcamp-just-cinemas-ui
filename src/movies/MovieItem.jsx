import React from "react";
import PropTypes from "prop-types";

const MovieItem = ({ name, slug, experiences }) => {
  const imageUrl = `https://img.spicinemas.in/resources/images/movies/${slug}/150x207.jpg`;
  return (
    <div className="movie-item">
      <img alt={name} className="movie-thumbnail" src={imageUrl} />
      <h5 className="name">{name}</h5>
      <p className="experiences">{experiences}</p>
    </div>
  );
};

MovieItem.defaultProps = {};

MovieItem.propTypes = {
  name: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired
};

export default MovieItem;
