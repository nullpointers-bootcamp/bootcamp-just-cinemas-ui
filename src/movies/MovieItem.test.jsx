import React from "react";
import renderer from "react-test-renderer";
import MovieItem from "./MovieItem";

it("show movie item", () => {
  const tree = renderer
    .create(<MovieItem name="name" slug="slug" experiences="experiences" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
