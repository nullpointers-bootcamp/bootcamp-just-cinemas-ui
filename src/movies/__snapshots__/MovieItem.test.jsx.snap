// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show movie item 1`] = `
<div
  className="movie-item"
>
  <img
    alt="name"
    className="movie-thumbnail"
    src="https://img.spicinemas.in/resources/images/movies/slug/150x207.jpg"
  />
  <h5
    className="name"
  >
    name
  </h5>
  <p
    className="experiences"
  >
    experiences
  </p>
</div>
`;
